### PR TITLE
[SYCL] Fix some Doxygen build warnings

### PR DESCRIPTION
**Pull request by vmaksimo**
This is done as a part of https://github.com/intel/llvm/issues/4140

* Changed several parameter types to parameter argument names (under \param tag)
* Renamed parameter names as they were changes in code, but not in documentation comment

### CODE REVIEW

**Comment by bader** [comment]
@vmaksimo, thank you for fixing this!

**Comment by vmaksimo** [comment]
> @vmaksimo, thank you for fixing this!

This is only the first step, but it should be a good start to fix the rest warnings :)

File: sycl/source/detail/scheduler/scheduler.hpp [hunk 1]
```diff
@@ -432,14 +432,14 @@ class Scheduler {
 
   /// Allocate buffers in the pool for a provided stream
   ///
-  /// \param Pointer to the stream object
-  /// \param Size of the stream buffer
-  /// \param Size of the flush buffer for a single work item
```
Comment (s-kanaev) [inline sycl/source/detail/scheduler/scheduler.hpp:437]: Will it really help without naming of arguments?

**Review by smaslov-intel — APPROVED** [review]

File: sycl/source/detail/scheduler/scheduler.hpp (hunk 1 above)
Comment (vmaksimo) [reply sycl/source/detail/scheduler/scheduler.hpp:437]: I believe it's better to leave the documentation with named arguments as it was planned by the author

**Pull request merged**